cmd/crank/beta/render/runtime_docker.go

```diff
@@ -191,58 +241,79 @@ func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
 
 	if r.PullPolicy == AnnotationValueRuntimeDockerPullPolicyAlways {
 		r.log.Debug("Pulling image with pullPolicy: Always", "image", r.Image)
-		err = PullImage(ctx, c, r.Image)
+		err = PullImage(ctx, cli, r.Image)
 		if err != nil {
-			return RuntimeContext{}, errors.Wrapf(err, "cannot pull Docker image %q", r.Image)
+			return "", "", errors.Wrapf(err, "cannot pull Docker image %q", r.Image)
 		}
 	}
 
 	// TODO(negz): Set a container name? Presumably unique across runs.
-	r.log.Debug("Creating Docker container", "image", r.Image, "address", addr)
-	rsp, err := c.ContainerCreate(ctx, cfg, hcfg, nil, nil, "")
+	r.log.Debug("Creating Docker container", "image", r.Image, "address", containerAddr, "name", r.Name)
+	rsp, err := cli.ContainerCreate(ctx, cfg, hcfg, nil, nil, r.Name)
 	if err != nil {
 		if !errdefs.IsNotFound(err) || r.PullPolicy == AnnotationValueRuntimeDockerPullPolicyNever {
-			return RuntimeContext{}, errors.Wrap(err, "cannot create Docker container")
+			return "", "", errors.Wrap(err, "cannot create Docker container")
 		}
 
 		// The image was not found, but we're allowed to pull it.
 		r.log.Debug("Image not found, pulling", "image", r.Image)
-		err = PullImage(ctx, c, r.Image)
+		err = PullImage(ctx, cli, r.Image)
 		if err != nil {
-			return RuntimeContext{}, errors.Wrapf(err, "cannot pull Docker image %q", r.Image)
+			return "", "", errors.Wrapf(err, "cannot pull Docker image %q", r.Image)
 		}
 
-		rsp, err = c.ContainerCreate(ctx, cfg, hcfg, nil, nil, "")
+		rsp, err = cli.ContainerCreate(ctx, cfg, hcfg, nil, nil, r.Name)
 		if err != nil {
-			return RuntimeContext{}, errors.Wrap(err, "cannot create Docker container")
+			return "", "", errors.Wrap(err, "cannot create Docker container")
 		}
 	}
+	if err := cli.ContainerStart(ctx, rsp.ID, container.StartOptions{}); err != nil {
+		return "", "", errors.Wrap(err, "cannot start Docker container")
+	}
+	return rsp.ID, containerAddr, errors.Wrap(err, "cannot start Docker container")
+}
 
-	if err := c.ContainerStart(ctx, rsp.ID, container.StartOptions{}); err != nil {
-		return RuntimeContext{}, errors.Wrap(err, "cannot start Docker container")
+// Start a Function as a Docker container.
+func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
+	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return RuntimeContext{}, errors.Wrap(err, "cannot create Docker client using environment variables")
+	}
+	containerAddr := ""
+	containerID := ""
+
+	if r.Name != "" {
+		// Check if the container is already running
+		containerID, containerAddr = r.findContainer(ctx, c, r.Name, true)
```

Comment:
Why pass `r.Name` as an argument? `r.findContainer` is a method on `r`, so it can already access `r.Name`.

It seems like the `startContainer` argument is always `true`, right? If so I'd suggest removing it.

Reply:
done